openlibrary/core/models.py

```diff
@@ -381,23 +392,58 @@ def from_isbn(cls, isbn: str, retry: bool = False):
         if isbn13 is None:
             return None  # consider raising ValueError
         isbn10 = isbn_13_to_isbn_10(isbn13)
+        isbns = [isbn13, isbn10] if isbn10 is not None else [isbn13]
 
         # Attempt to fetch book from OL
-        for isbn in [isbn13, isbn10]:
+        for isbn in isbns:
             if isbn:
                 matches = web.ctx.site.things(
                     {"type": "/type/edition", 'isbn_%s' % len(isbn): isbn}
                 )
                 if matches:
                     return web.ctx.site.get(matches[0])
 
-        # Attempt to create from amazon, then fetch from OL
-        retries = 5 if retry else 0
-        key = (isbn10 or isbn13) and create_edition_from_amazon_metadata(
-            isbn10 or isbn13, retries=retries
-        )
-        if key:
-            return web.ctx.site.get(key)
+        # Attempt to fetch the book from the import_item table
+        if result := ImportItem.find_staged_or_pending(identifiers=isbns):
+            item: ImportItem = ImportItem(result[0])
+
+            try:
+                # Avoids a circular import issue.
+                from openlibrary.plugins.importapi.code import parse_data
```

Comment:
This code needs to get out of the generic models and the /api/import path and needs to be contained either in **catalog/addbook/__init__.py** (or w/e) or in core/imports (which seems less relevant because that really deals with the batch import system, not imports generally)
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/importapi/code.py#L132-L142